eng/common/TestResources/deploy-test-resources.yml

```diff
@@ -43,6 +43,18 @@ steps:
   - template: /eng/common/TestResources/setup-environments.yml
 
   - ${{ if eq('true', parameters.UseFederatedAuth) }}:
+    - task: AzureCLI@2
```

Comment:
Do you know if there is a way to get similar values in the PS context? Also should we make this an opt-in as I'd hate for everyone to have to spend time logging into Az CLI and Az Powershell and not even end up needing these values.

Reply:
From what I could tell there is not. It's really unfortunate.

I agree re: opt-in. I can add the parameter plumbing.

Reply:
Couldn't we call Get-AzAccessToken to get another token to use with az cli?

Reply:
Looks like the JWT is coming from a helper library from the service connection https://github.com/microsoft/azure-pipelines-tasks/blob/63938db07a732c279ace3a71b6da3321db636d1a/Tasks/AzurePowerShellV5/azurepowershell.ts#L85

Reply:
I can't find any of the inner source for that call :(

Reply:
You can definitely get the client id and tenant from https://github.com/microsoft/azure-pipelines-tasks/blob/master/Tasks/AzurePowerShellV5/InitializeAz.ps1#L121 the question is how do you get a token. I'm guessing we could call Get-AzAccessToken to get one as needed depending on the resource type.